lib/semian/circuit_breaker.rb

```diff
@@ -30,9 +32,7 @@ def acquire
     end
 
     def request_allowed?
-      return true if closed?
-      half_open if error_timeout_expired?
-      !open?
+      closed? || half_open? || (open? && error_timeout_expired?)
```

Comment:
> Or should that condition even just go in half_open? to begin with?

It can't really go in `half_open?` because then you wouldn't know if you have to make a transition or not. So yeah that piece isn't necessarily trivial to understand. I'm usually not a comment guy but it's likely worth it here.